Source/Provers/SMTLib/SMTLibInteractiveTheoremProver.cs

```diff
@@ -41,6 +41,19 @@ public override Task GoBackToIdle()
       return Process.PingPong();
     }
 
+    public void ClearProverErrors()
+    {
+      lock (proverErrors)
```

Comment:
Since `proverErrors` and `proverWarnings` are always locked before they're modified, we might as well use a concurrent data structure.

Reply:
In think this is equivalent, but in either case I'm only worried about the following scenario, in which we empty the list, release the lock, and then the previous process adds one more error before we restart the process.

Reply:
To avoid the problem I mention, I am attaching the proverErrors and proverWarnings to the Process field through a ConditionalWeakTable. 

Moreover, I changed their types to ConcurrentStack so that there is no need for locking mechanisms; however, when they are invoked once in a conditional statement, and once in the body of that conditional statement, I ensured that it's the same object we are considering instead of accessing the field twice.

Can you have a look to my new solution?